lib/internal/modules/run_main.js

```diff
@@ -64,8 +68,7 @@ function runMainESM(mainPath) {
   const { pathToFileURL } = require('internal/url');
 
   handleMainPromise(loadESM((esmLoader) => {
-    const main = path.isAbsolute(mainPath) ?
-      pathToFileURL(mainPath).href : mainPath;
+    const main = getOptionValue('--experimental-entry-url') ? mainPath : pathToFileURL(mainPath).href;
```

Comment:
Absolute `https:` and `data:` URLs should work too. `node --entry-url data:text/javascript,console.log(3)` should work.